tools/run_tests/xds_k8s_test_driver/tests/bootstrap_generator_test.py

```diff
@@ -0,0 +1,216 @@
+                                            attempts=3,
+                                            log_level=logging.INFO)
+        try:
+            retryer(self._cleanup)
+        except retryers.RetryError:
+            logger.exception('Got error during teardown')
+        super().tearDown()
+
+    def _cleanup(self):
+        self.client_runner.cleanup(force=self.force_cleanup)
+        self.removeServerBackends()
+        self.server_runner.cleanup(force=self.force_cleanup)
+
+    @parameterized.named_parameters(
+        # Add images corresponding to future releases here.
+        #
+        # TODO: Update bootstrap generator release instructions to add a newly
+        # released version to this list.
+        ('v0.14.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:d6baaf7b0e0c63054ac4d9bedc09021ff261d599'
+        ),
+        ('v0.13.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:203db6ce70452996f4183c30dd4c5ecaada168b0'
+        ),
+        ('v0.12.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:8765051ef3b742bc5cd20f16de078ae7547f2ba2'
+        ),
+        ('v0.11.0',
+         'gcr.io/grpc-testing/td-grpc-bootstrap:b96f7a73314668aee83cbf86ab1e40135a0542fc'
+        ),
+        # v0.10.0 uses v2 xDS transport protocol by default. TD only supports v3
+        # and we can force the bootstrap generator to emit config with v3
+        # support by setting the --include-v3-features-experimental flag to
+        # true.
+        #
+        # TODO: Figure out how to pass flags to the bootstrap generator via the
+        # client and server runners, and uncomment this version.
+        # ('v0.10.0', 'gcr.io/grpc-testing/td-grpc-bootstrap:66de7ea0e170351c9fae17232b81adbfb3e80ec3'),
+    )
+    def test_baseline_in_server_with_bootstrap_version(self, image):
+        """Runs the baseline test for multiple versions of the bootstrap
+        generator on the server.
+        """
+        self.server_runner = self.initKubernetesServerRunner(
+            td_bootstrap_image=image)
+        self.test_server = BootstrapGeneratorClientTest.startTestServer(
+            server_runner=self.server_runner,
+            port=self.server_port,
+            maintenance_port=self.server_maintenance_port,
+            xds_host=self.server_xds_host,
+            xds_port=self.server_xds_port)
+
+        # Load backends.
+        neg_name, neg_zones = self.server_runner.k8s_namespace.get_service_neg(
+            self.server_runner.service_name, self.server_port)
+
+        # Add backends to the Backend Service.
+        self.td.backend_service_add_neg_backends(neg_name, neg_zones)
+        self.td.wait_for_backends_healthy_status()
+
+        self.client_runner = self.initKubernetesClientRunner()
+        test_client: XdsTestClient = self.startTestClient(self.test_server)
+        self.assertXdsConfigExists(test_client)
+        self.assertSuccessfulRpcs(test_client)
+
+
+if __name__ == '__main__':
+    absltest.main(failfast=True)
```

Comment:
You sure you want to bail out on the first fail? If no:
```suggestion
    absltest.main()
```